src/shifter_core.c

```diff
@@ -3151,7 +3151,21 @@ int unmountTree(MountList *mounts, const char *base) {
     setSort_MountList(mounts, MOUNT_SORT_REVERSE);
 
     for (ptr = mounts->mountPointList; ptr && *ptr; ptr++) {
+        size_t len = strlen(*ptr);
         if (strncmp(*ptr, base, baseLen) == 0) {
+            char *next_slash = strchr(*ptr + baseLen, '/');
+
+            /* avoid unmounting a path for which the base is just a substring
+             * base: /var/udiMount/cvmfs
+             * *ptr: /var/udiMount/cvmfs_nfs/123  <-- case 1, not ok to unmount
+             * *ptr: /var/udiMount/cvmfs_nfs      <-- case 2, not ok to unmount
+             */
+            if (next_slash && next_slash - *ptr > baseLen) {
```

Comment:
The logic in this section is pretty subtle.  Would it make sense to make a function for this so tests could be added for the cases you outline in the comment above?

Reply:
those tests have been added in 01eb311